spec/classes/collectd_init_spec.rb

```diff
@@ -9,20 +9,13 @@
 
       options = os_specific_options(facts)
       context 'with all defaults' do
-        it { is_expected.to contain_class('collectd') }
-        it { is_expected.to contain_class('collectd::params') }
         it { is_expected.to compile.with_all_deps }
-        it { is_expected.to contain_anchor('collectd::begin') }
-        it { is_expected.to contain_anchor('collectd::end') }
         it { is_expected.to contain_file('collectd.conf').without_content }
         it { is_expected.to contain_file('collectd.d').with_ensure('directory') }
         it { is_expected.to contain_file_line('include_conf_d').with_ensure('absent') }
         it { is_expected.to contain_file_line('include_conf_d_dot_conf').with_ensure('present') }
         it { is_expected.to contain_package(options[:package]).with_ensure('present') }
         it { is_expected.to contain_package(options[:package]).with_install_options(nil) }
-        it { is_expected.to contain_class('collectd::install').that_comes_before('Class[collectd::config]') }
```

Comment:
those checks should still be valid, since we use ->?

Reply:
Valid, but again I don't think it is useful, as we don't have any conditional logic which will result in the order of classes changing.